src/utils/random-data.js

```diff
@@ -34,13 +35,38 @@ const LAYERS = [
  */
 class Pipeline {
   constructor() {
+    this.seedRandomData();
     this.rankCount = this.getRankCount();
     this.rankLayers = this.getRankLayers();
     this.tags = this.generateTags();
     this.nodes = this.generateNodes();
     this.edges = this.generateEdges();
   }
 
+  /**
+   * Seed data with a random hash, allowing it to be reproduced.
+   * If the URL searchParams contain a 'seed' key then use its value,
+   * else create a new one, and make it available via the console.
+   */
+  seedRandomData() {
+    if (typeof window === 'undefined') {
+      return;
+    }
+    const url = new URL(window.location);
+    let seed = url.searchParams.get('seed');
+    if (!seed) {
+      seed = generateHash(30);
+      url.searchParams.set('seed', seed);
+    }
+    if (typeof jest === 'undefined') {
+      console.info(
+        `%cRandom data seed: ${seed}\nTo reuse this layout, visit ${url.toString()}`,
+        'font-weight: bold'
+      );
+    }
+    seedrandom(seed, { global: true });
```

Comment:
I think it's better to use the seeded random function directly rather than patching the global random, as not to potentially clash with any other uses on the page.